llvm/unittests/Target/AIE/VirtUnrollAliasAnalysisTest.cpp

```diff
@@ -0,0 +1,254 @@
+        MemoryBuffer::getMemBuffer(MIRString);
+    Parser = createMIRParser(std::move(MBuffer), Context);
+    if (!Parser)
+      report_fatal_error("null MIRParser");
+    M = Parser->parseIRModule();
+    if (!M)
+      report_fatal_error("parseIRModule failed");
+    M->setTargetTriple(TM->getTargetTriple().getTriple());
+    M->setDataLayout(TM->createDataLayout());
+    MMI = std::make_unique<MachineModuleInfo>(TM.get());
+    if (Parser->parseMachineFunctions(*M, *MMI.get()))
+      report_fatal_error("parseMachineFunctions failed");
+  }
+
+  MachineFunction *getMachineFunction(Module *M, StringRef Name) {
+    auto F = M->getFunction(Name);
+    if (!F)
+      report_fatal_error("null Function");
+    auto &MF = MMI->getOrCreateMachineFunction(*F);
+    return &MF;
+  }
+
+  void getInstructions(const MachineBasicBlock *MBB,
+                       std::vector<const MachineInstr *> &Instr) {
+    for (const MachineInstr &MI : *MBB) {
+      Instr.push_back(&MI);
+    }
+  }
+};
+
+TEST_F(VistUnrollAliasAnalysisTest, TestSameIterators) {
+  const MachineFunction *F = getMachineFunction(M.get(), "test");
+  const MachineBasicBlock *MBB = F->getBlockNumbered(2);
+  std::vector<const MachineInstr *> Instr;
+  getInstructions(MBB, Instr);
+  const MachineInstr *Load1 = Instr[0];
+  const MachineInstr *Store1 = Instr[4];
+  const MachineInstr *Load2 = Instr[5];
+  const MachineInstr *Store2 = Instr[6];
+  const MachineInstr *Load3 = Instr[7];
+  const MachineInstr *Store3 = Instr[8];
+
+  // Same load, same virt unroll.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load1, Load1, 0, 0) ==
+              AliasResult::MayAlias);
+  // Same load, same virt unroll (3rd iteration).
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load1, Load1, 2, 2) ==
+              AliasResult::MayAlias);
+  // Same load, diff. virt unroll.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load1, Load1, 0, 1) ==
+              AliasResult::NoAlias);
+  // Diff. load, same virt unroll.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load1, Load2, 0, 0) ==
+              AliasResult::NoAlias);
+  // Store and load with same virt unroll, same pointer step.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load1, Store1, 0, 0) ==
+              AliasResult::MayAlias);
+  // Store and load with same virt unroll, same pointer step.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load2, Store2, 0, 0) ==
+              AliasResult::MayAlias);
+  // Store and load with diff virt unroll, same pointer step.
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Load3, Store3, 4, 0) ==
+              AliasResult::NoAlias);
+  // Store and load with diff virt unroll, diff pointer step.
+  // This case is interesting because the first store from the 3rd iteration
+  // will alias with the 3rd load of the second iteration since we are not
+  // updating the pointer after the last load and store (post. inc.).
+  ASSERT_TRUE(AIE::aliasAcrossVirtualUnrolls(Store1, Load3, 3, 2) ==
```

Comment:
Nit: `EXPECT_TRUE` is better because it doesn't stop the test execution at the first failure.

Reply:
Interesting, I was now aware! I think I will leave as is this time to not interrupt CI again ;-).

Reply:
I considered this! Already updated.